workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/GameObjectRepresentation/SpatialOSBehaviourLibraryTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.CodeDom;
+using System.Linq;
+using System.Text.RegularExpressions;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Object = UnityEngine.Object;
+
+namespace Improbable.Gdk.Core.EditmodeTests
+{
+    [TestFixture]
+    public class SpatialOSBehaviourLibraryTests
+    {
+        private const uint ComponentId1 = 1337;
+        private const uint ComponentId2 = 1338;
+
+        [ReaderInterface]
+        [ComponentId(ComponentId1)]
+        private interface DummyReader
+        {
+        }
+
+        [WriterInterface]
+        [ComponentId(ComponentId1)]
+        private interface DummyWriter1
+        {
+        }
+
+        [WriterInterface]
+        [ComponentId(ComponentId2)]
+        private interface DummyWriter2
+        {
+        }
+
+        private class SingleReaderBehaviour : MonoBehaviour
+        {
+            [Require] public DummyReader Reader;
+        }
+
+        private class TwoWritersBehaviour : MonoBehaviour
+        {
+            [Require] public DummyWriter1 Writer1;
+            [Require] public DummyWriter2 Writer2;
+        }
+
+        private class MultipleReadersOfSameType : MonoBehaviour
+        {
+            [Require] public DummyReader Reader1;
+            [Require] public DummyReader Reader2;
```

Comment:
If we insist that we do need support for injecting properties, then we need tests for them as well.

Reply:
Done